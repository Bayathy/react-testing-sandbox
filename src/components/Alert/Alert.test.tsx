import { cleanup, render } from "@testing-library/react"

import * as AlertContext from "~/context/AlertContext"

import { Alert } from "./"

let alertStateSpy: jest.SpyInstance<unknown>

describe("Alert", () => {
  beforeEach(() => {
    alertStateSpy = jest.spyOn(AlertContext, "useAlertState")
    alertStateSpy.mockImplementation(() => ({
      show: false,
      message: ""
    }))
  })
  afterEach(() => {
    alertStateSpy.mockClear()
    cleanup()
  })
  test("render: null", () => {
    const { container } = render(<Alert />)
    expect(container.innerHTML).toBe("")
  })
  test("render: context update", () => {
    alertStateSpy.mockImplementation(() => ({
      show: true,
      message: "test message"
    }))
    const { asFragment } = render(<Alert />)
    expect(asFragment()).toMatchSnapshot()
  })
})
