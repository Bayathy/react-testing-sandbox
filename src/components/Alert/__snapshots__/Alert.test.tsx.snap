// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert render: context update 1`] = `
<DocumentFragment>
  <div
    style="background: red; padding: 10px;"
  >
    test message
  </div>
</DocumentFragment>
`;
